GeneratorInterface/LHEInterface/scripts/mergeLHE.py

```diff
@@ -2,10 +2,149 @@
+            new_init_block[ipr][1] = math.sqrt(new_init_block[ipr][1]) / tot_nevent
+            print('  xsecup, xerrup, xmaxup, lprup: %.6E, %.6E, %.6E, %d' % tuple(new_init_block[ipr] + [ipr]))
+            self._merged_init_str += '%.6E %.6E %.6E %d\n' % tuple(new_init_block[ipr] + [ipr])
+        self._merged_init_str += '\n'.join(info_after_subprocess)
+        if len(info_after_subprocess):
+            self._merged_init_str += '\n'
+
+        return self._merged_init_str
+
+    def merge(self):
+        with open(self.outputFileName, 'w') as fw:
+            # Read the header for the first file then write to the output
+            while True:
+                line = next(self._f[0])
+                if line.startswith('<init>'):
+                    break
+                else:
+                    fw.write(line)
+
+            # Let all inputFiles reach the start of <init> block
+            for i in range(1, len(self._f)):
+                while(not next(self._f[i]).startswith('<init>')):
+                    pass
+
+            # Read <init> blocks for all inputFiles
+            for i in range(len(self._f)):
+                init_str = ''
+                while True:
+                    line = next(self._f[i])
+                    if line.startswith('</init>'):
+                        break
+                    else:
+                        init_str += line
+                self._init_str.append(init_str)
+
+            # Iterate over all events file-by-file and write events temporarily to .tmp.lhe
+            with open('.tmp.lhe', 'w') as _fwtmp:
+                for i in range(len(self._f)):
+                    nevent = 0
+                    while True:
+                        line = next(self._f[i])
+                        if line.startswith('</event>'):
+                            nevent += 1
+                        if line.startswith('</LesHouchesEvents>'):
+                            break
+                        else:
+                            _fwtmp.write(line)
+                    self._nevent.append(nevent)
+                    self._f[i].close()
+
+            # Merge the init blocks and write to the output
+            fw.write('<init>\n' + self.merge_init_blocks() + '</init>\n')
+
+            # Write event blocks in .tmp.lhe back to the output
+            with open('.tmp.lhe', 'r') as _ftmp:
+                for _line in _ftmp:
+                    fw.write(_line)
+            fw.write('</LesHouchesEvents>\n')
+            os.remove('.tmp.lhe')
+
+
+class MadGraph5LHEMerger(BaseLHEMerger):
+    """Use the merger script in genproductions dedicated for MadGraph5-produced LHEs"""
+    def __init__(self, inputFiles, outputFileName):
+        super(MadGraph5LHEMerger, self).__init__(inputFiles, outputFileName)
+        self._merger_script_url = 'https://raw.githubusercontent.com/cms-sw/genproductions/5c1e865a6fbe3a762a28363835d9a804c9cf0dbe/bin/MadGraph5_aMCatNLO/Utilities/merge.pl'
```

Comment:
Shouldn't it be https://raw.githubusercontent.com/cms-sw/genproductions/master/bin/MadGraph5_aMCatNLO/Utilities/merge.pl ?

Reply:
Yes this is a consideration of stability. We use a fixed version of the script.